dashboard/src/components/AppView/AppView.tsx

```diff
@@ -204,20 +204,22 @@ export default function AppView() {
       cluster,
       app.installedPackageRef?.context?.namespace || "",
     );
-    setResourceRefs(parsedRefs);
+    setAppViewResourceRefs(parsedRefs);
     return () => {};
-  }, [app, cluster, kinds, resourceRefs]);
+  }, [app, cluster, kinds, appViewResourceRefs]);
 
   useEffect(() => {
-    Object.values(resourceRefs).forEach((refs: ResourceRef[]) => {
-      refs.forEach(ref => dispatch(actions.kube.getAndWatchResource(ref)));
-    });
+    if (!app || !app.installedPackageRef || !app.apiResourceRefs) {
+      return () => {};
+    }
+    // TODO(minelson): Update to watch only those resources that
+    // we're interested in watching change (deployments, statefulsets etc.)
+    // and get the rest.
+    dispatch(actions.kube.getResources(app!.installedPackageRef!, app!.apiResourceRefs, true));
     return function cleanup() {
```

Comment:
Nice! btw, we are not usually using cleanup functions. Perhaps at some point, we should revisit it to ensure we don't have any resource left (I recall to have seen some unmount warnings at the console)

Reply:
Yeah, it's required here, because unmounting this component triggers the action to finish the request (and remove the subscription from the state, which reminds me, I want to remove it from the state as it's no longer necessary to track there. I'll do that in my current branch).